tensorflow/lite/micro/kernels/broadcast_to.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#include "tensorflow/lite/kernels/internal/reference/broadcast_to.h"
+
+#include <stdint.h>
+
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/micro_context.h"
+
+namespace tflite {
+
+namespace {
+constexpr int kInputTensor = 0;
+constexpr int kShapeTensor = 1;
+constexpr int kOutputTensor = 0;
+// A maximum of 5 dimensions.
+constexpr int kMaxDims = 5;
+
+struct BroadcastToContext {
+  BroadcastToContext(TfLiteContext* context, TfLiteNode* node) {
+    micro_context = GetMicroContext(context);
+    input = micro_context->AllocateTempInputTensor(node, kInputTensor);
```

Comment:
I have get rid of this structure/class completely. PTAL.

The trouble is that class constructor is not ideal in handling error. The pattern to address initialization error that I knows of typically requires injection for any things that can fail at initialization. I feel this is a bit loaded for this goal.
This structure is carried from lite's implementation to keep implementation closed to the lite's implementation, but it is not too bad to get rid of it. So that is what I ended up doing.

Reply:
Changes look good. It looks like the struct is still here, perhaps missed removing it?

Reply:
Thanks for catching this! Complete the change now.